### PR TITLE
CARBON-888 add package-lock file

### PR DESCRIPTION
**Pull request by abramin**
# Description

* to allow package locks when using npm 5
* Refactor date-range to satisfy compiler issue on upgrade.

Had to run `npm rebuild node-sass` after reinstall.
* Also needed to run `npm verify cache` before npm install in carbon root.

### CODE REVIEW

**Comment by abramin** [comment]
@ianoxley Possibly, everything seemed to work without the change, but will do that.

File: check-version.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ var exec = require('child_process').exec;
 exec("npm -v", function(err, stdout, stderr) {
   if (parseFloat(stdout) < 4) {
     console.error("Using npm version: " + stdout);
-    console.error("You need to be using at least npm version 4. To upgrade use `npm install npm@4.6.1 -g`.\n");
+    console.error("You need to be using at least npm version 4. To upgrade use `npm install npm@5.3.0 -g`.\n");
```
Comment (ianoxley) [inline check-version.js]: Are we still supporting npm 4 as well? Might that cause problems with the package-lock.json file?

Also, if we use "`npm install npm@latest`" we won't have to keep updating this script with every npm 5 release.

Comment (abramin) [inline check-version.js]: Considered latest - happy to change that. People can still use  4 - the package.lock will just get ignored.

Comment (ianoxley) [inline check-version.js]: npm 5 had quite a few breaking changes though, so it might be better to drop support for v4.

**Comment by abramin** [comment]
@andrewjtait Tried install after verifying cache, all good 👍

**Pull request merged**